### PR TITLE
Search

### PR DESCRIPTION
**Pull request by tobybatch**
I've merged in Israt's changes, fixed the broken component loads, and the CRSF token issue.

### CODE REVIEW

File: cms/search/templates/search/search.html [hunk 1]
```diff
@@ -8,7 +8,12 @@
 {% block content %}
   <div class="nhsuk-grid-row">
     <div class="nhsuk-grid-column-full">
-      <h1 class="nhsuk-heading-l">Search results for: Condition</h1>
+      <h1 class="nhsuk-heading-l">
+        Search results for:
+        {% if query_string %}
```
Comment (jacksonj04) [inline cms/search/templates/search/search.html:13]: I think this `if` statement is in the wrong place - if there isn't a query we'll end up with "Search results for:" and nothing after it - maybe wrap this around the entire search string block?

File: cms/search/templates/search/search.html [hunk 2]
```diff
@@ -8,7 +8,12 @@
...
+          {{ query_string }}{% if content_type %}, filtered by <strong>{{ content_type }}</strong>{% endif %}
```
Comment (jacksonj04) [inline cms/search/templates/search/search.html:14]: Should `{{ query_string }}` also be wrapped in a `<strong>`?

File: cms/search/templates/search/search.html [hunk 3]
```diff
@@ -20,34 +25,42 @@ <h2 class="nhsuk-heading-xs"></h2>
...
   <div class="nhsuk-grid-row nhsuk-u-margin-top-8">
-    <div class="nhsuk-grid-column-one-third">
-      <h2 class="nhsuk-heading-xs">Filters</h2>
-      {% include 'partials/nhs_components/search_content_type.html' %}
-    </div>
-    {# body#}
-    <div class="nhsuk-grid-column-two-thirds">
-      <div class="search-results-heading">
-        {% include 'partials/nhs_components/select_search_results_by_order.html' %}
+  <div class="nhsuk-grid-column-one-third">
```
Comment (jacksonj04) [inline cms/search/templates/search/search.html:28]: This looks like an unexpected indentation change which makes the diff bigger than it should be.

File: cms/search/views.py [hunk 4]
```diff
@@ -9,6 +9,8 @@
 from cms.blogs.models import Blog
 from cms.pages.models import BasePage
 
+RESULTS_PER_PAGE = 10
```
Comment (jacksonj04) [inline cms/search/views.py:12]: Can this live in `base.py` so we keep all the settings in one place, then:

``` python
from django.conf import settings
```

File: cms/search/views.py [hunk 5]
```diff
@@ -63,7 +65,7 @@ def search(_class):
         search_results = Page.objects.none()
 
     # Pagination
-    paginator = Paginator(search_results, 10)
+    paginator = Paginator(search_results, RESULTS_PER_PAGE)
```
Comment (jacksonj04) [inline cms/search/views.py:68]: Assuming `RESULTS_PER_PAGE` moves:

```suggestion
    paginator = Paginator(search_results, settings.RESULTS_PER_PAGE)
```

File: cms/templates/partials/nhs_components/promoted_search-content.html [hunk 6]
```diff
@@ -1,16 +1,17 @@
-<div class="nhsuk-notification-banner" role="region" aria-labelledby="nhsuk-notification-banner-title" data-module="nhsuk-notification-banner">
+{% load wagtailcore_tags wagtailsearchpromotions_tags %}
+<div class="nhsuk-notification-banner" role="region" aria-labelledby="nhsuk-notification-banner-title"
+     data-module="nhsuk-notification-banner">
   <div class="nhsuk-notification-banner__header">
     <h2 class="nhsuk-notification-banner__title" id="nhsuk-notification-banner-title">
-    Suggest result
+      Suggest result
```
Comment (jacksonj04) [inline cms/templates/partials/nhs_components/promoted_search-content.html:6]:
```suggestion
      Suggested result
```

File: cms/templates/partials/nhs_components/search_results_bar.html [hunk 7]
```diff
@@ -1,10 +1,11 @@
 {# TODO developers to make search work #}
 <div class="nhs-search">
-  <form action="#" method="post">
+  <form action="#" method="get">
+    {% csrf_token %}
     <div class="nhsuk-form-group">
       <label class="nhsuk-label nhs-search__label nhsuk-!-font-weight-bold" for="search">
       </label>
-      <input class="nhsuk-input nhs-search__input " id="search" name="search" type="search" aria-describedby="search-hint">
+      <input class="nhsuk-input nhs-search__input " id="search" name="query" type="search" value="{{query_string}}" aria-describedby="search-hint">
```
Comment (jacksonj04) [inline cms/templates/partials/nhs_components/search_results_bar.html:8]: I think this is missing a `required`.

File: cms/templates/partials/nhs_components/select_search_results_by_order.html [hunk 8]
```diff
@@ -1,3 +1,4 @@
+{# #}
```
Comment (jacksonj04) [inline cms/templates/partials/nhs_components/select_search_results_by_order.html:1]: Errant comment!

**Review by jacksonj04 — CHANGES_REQUESTED** [review]
A few tweaks, and some unexpected whitespace change which makes this diff look bigger than it is.

File: cms/search/templates/search/search.html (hunk 1 above)
Comment (dragon-dxw) [reply cms/search/templates/search/search.html:12]: > "Search results for: kittens"
Sorry: what if "kittens" is instead just the empty string, is "Search results for: " then nothing acceptable?

> thats correct
dont show

File: cms/settings/base.py [hunk 9]
```diff
@@ -305,3 +305,5 @@
 NHSEI_MAX_MENU_CAPTION_LENGTH = 70
 # TODO - turn this on when we go live, it breaks imports all ove the place
 WAGTAILREDIRECTS_AUTO_CREATE = False
+# pagination page size for search results
+RESULTS_PER_PAGE = 10
```
Comment (dragon-dxw) [inline cms/settings/base.py:309]: Having improved it beyond how I'd left it, I'm sorry -- now it's moved, is it worth being explicitly `SEARCH_RESULTS_PER_PAGE`?

File: cms/templates/partials/nhs_components/select_search_results_by_order.html [hunk 10]
```diff
@@ -16,6 +15,9 @@ <h1 class="nhsuk-fieldset__heading">Sort by:</h1>
           <option value="-latest_revision_created_at"{% if order == '-latest_revision_created_at' %} selected{% endif %}>Most relevant</option>
           <option value="latest_revision_created_at"{% if order == 'latest_revision_created_at' %} selected{% endif %}>Newest first</option>
           <option value="-first_published_at"{% if order == '-first_published_at' %} selected{% endif %}>Oldest first</option>
+          <option value="most_relevant"{% if order == "most_relevant" %} selected{% endif %}>Most relevant</option>
```
Comment (dragon-dxw) [inline cms/templates/partials/nhs_components/select_search_results_by_order.html:18]: This is almost certainly a placeholder which I was working on just before I dropped it -- we probably need some code to make whatever this value is not sort the search results at all (and trust ElasticSearch returns them in a sensible order.)

This almost certainly currently breaks due to not being a valid search order.

My plan was to detect this value and not `.order_by()` the results if encountered in templates/search/views.py

File: cms/search/views.py (hunk 5 above)
Comment (tobybatch) [reply cms/search/views.py]: Done

File: cms/search/templates/search/search.html (hunk 3 above)
Comment (tobybatch) [reply cms/search/templates/search/search.html:30]: Yep, the HTML indents are causing the big diff.

File: cms/templates/partials/nhs_components/search_results_bar.html (hunk 7 above)
Comment (tobybatch) [reply cms/templates/partials/nhs_components/search_results_bar.html]: Done

File: cms/templates/partials/nhs_components/select_search_results_by_order.html (hunk 10 above)
Comment (tobybatch) [reply cms/templates/partials/nhs_components/select_search_results_by_order.html:18]: Can we raise this as a new backlog ticket, to fix ordering?

File: cms/search/templates/search/search.html (hunk 1 above)
Comment (tobybatch) [reply cms/search/templates/search/search.html:12]: https://dxw.slack.com/archives/C027VJH9S3A/p1649329060036969?thread_ts=1649328175.241669&cid=C027VJH9S3A

Fix on the way

File: cms/templates/partials/nhs_components/select_search_results_by_order.html (hunk 10 above)
Comment (dragon-dxw) [reply cms/templates/partials/nhs_components/select_search_results_by_order.html:18]: Agreed! Can you link/copypaste that text into the ticket?

**Review by dragon-dxw — APPROVED** [review]
Agreed that we should leave the search order for another day. These are good improvements!

**Review by jacksonj04 — APPROVED** [review]
Ship it!

**Pull request merged**